File: java/playwright/src/test/java/dev/selenium/request/ModifyMethodTest.java
package dev.selenium.request;

import com.microsoft.playwright.Route;
import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ModifyMethodTest extends TestBase {

  @Test
  public void modifyHttpMethod() {
    page.route(
        "https://selenium.dev/**",
        route -> {
          route.resume(new Route.ResumeOptions().setMethod("HEAD"));
        });

    page.navigate("https://selenium.dev");
    Assertions.assertEquals("", page.textContent("body"));
  }
}

File: dotnet/Playwright/Request/ModifyMethodTest.cs
using Microsoft.Playwright;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright.Request;

[TestClass]
public class ModifyMethodTest : TestBase
{
    [TestMethod]
    public async Task ModifyHttpMethod()
    {
        await Page!.RouteAsync("https://selenium.dev/**", async route =>
        {
            await route.ContinueAsync(new RouteContinueOptions { Method = "HEAD" });
        });

        await Page.GotoAsync("https://selenium.dev");
        var bodyText = await Page.TextContentAsync("body");
        Assert.AreEqual("", bodyText);
    }
}